lib/Traits/TraitsClasses/ApiTableCatalogItemTraits.ts

```diff
@@ -29,12 +30,28 @@ export class ApiTableRequestTraits extends mixTraits(ApiRequestTraits) {
   columnMajorColumnNames?: string[] = ["value"];
 }
 
+export class ApiTableColumnTraits extends ModelTraits {
+  @primitiveTrait({
+    name: "Name",
+    type: "string",
+    description: "Column name. This must match name in `columns`."
+  })
+  name?: string;
+
+  @primitiveTrait({
+    name: "Response data path",
+    type: "string",
+    description:
+      "Path to relevant data in JSON response. eg: `some.user.name`, `some.users[0].name` or `some.users[].name`. For data to be parsed correctly, it must return a primitive value (string, number, boolean)."
+  })
+  responseDataPath?: string;
+}
+
 export default class ApiTableCatalogItemTraits extends mixTraits(
   TableTraits,
   CatalogMemberTraits,
   LegendOwnerTraits,
-  AutoRefreshingTraits,
-  ApiRequestTraits
```

Comment:
Note I have removed `ApiRequestTraits` here because `ApiTableCatalogItem` only supported `queryParameters` and `updateQueryParameters` - I have re-added this manually